services/core/MasterDriverAgent/master_driver/interfaces/ecobee.py

```diff
@@ -0,0 +1,851 @@
+                                     headers, update_frequency=180, params=params, refresh=refresh).get()
+            if data is None:
+                raise RuntimeError("No Ecobee data available from HTTP Proxy Agent.")
+            _log.info("Last Ecobee data update occurred: {}".format(data.get("request_timestamp")))
+            self.ecobee_data = data.get("request_response")
+        except RemoteError:
+            if retry:
+                _log.warning("Failed to get Ecobee data from HTTP Proxy, refreshing tokens and trying again.")
+                self.refresh_tokens()
+                self.get_ecobee_data(refresh=refresh, retry=False)
+            else:
+                raise RuntimeError("Failed to get Ecobee data from HTTP Proxy after refreshing tokens. May be "
+                                   "experiencing connection issues or Ecobee API may be down.")
+
+    def get_point(self, point_name, **kwargs):
+        """
+        Return a point's most recent stored value from remote API
+        :param point_name:
+        :return:
+        """
+        # Find the named register and get its current state from the periodic Ecobee API data
+        register = self.get_register_by_name(point_name)
+        try:
+            return register.get_state(self.ecobee_data)
+        except:
+            self.refresh_tokens()
+            return register.get_state(self.ecobee_data)
+
+    def _set_point(self, point_name, value, **kwargs):
+        """
+        Send request to remote API to update a point based on provided parameters
+        :param point_name: Name of the point to update
+        :param value: Intended update value
+        :return: Updated state from remote API
+        """
+        # Find the correct register by name, set its state, then fetch the new state based on the register's type
+        register = self.get_register_by_name(point_name)
+        if register.read_only:
+            raise IOError("Trying to write to a point configured read only: {}".format(point_name))
+        if not register.register_type in ["setting", "hold", "vacation", "programs"]:
+            raise RuntimeError("Register {} type {} does not support set_point".format(register.point_name,
+                                                                                       register.register_type))
+        try:
+            if register.register_type == "setting" or register.register_type == "hold":
+                register.set_state(value)
+            elif register.register_type in ["vacation", "programs"]:
+                register.set_state(value, **kwargs)
+        except:
+            self.refresh_tokens()
+            if register.register_type == "setting" or register.register_type == "hold":
+                register.set_state(value)
+            elif register.register_type in ["vacation", "programs"]:
+                register.set_state(value, **kwargs)
```

Comment:
In the case that the exception is thrown again, the driver should thrown an error as it's unable to fetch remote data.

Reply:
But you not catching the exception the second time

Reply:
I do not want the exception to be caught the second time. At this point the data is not accessible, so it would be expected that an exception be thrown in setting the state.